boards: stm32f746g_disco ft5336 support, plus sample application

k0d opened the pull request:
The first commit adds support for the ft5336 touch panel on the stm32f746g_disco board. It enables the driver and adds the relevant device tree config. It's dependent on the I2C_3 device so this is enabled also.

The second commit adds a sample application so it can be tested. It's tested on the stm32f746g_disco board, but should work on other boards for example the mimxrt1060_evk.

k0d committed:
boards: Configure ft5336 touch panel on stm32f746g_disco board

Adds device tree nodes, pinmux definitions and configures Kconfig
defaults for the ft5336 touch panel driver on the
stm32f746g_disco board.

Signed-off-by: Mark Olsson <mark@markolsson.se>

k0d commented:
@MaureenHelm Could you test the sample on the mimxrt1060_evk board please?

'zephyr/samples/drivers/kscan_ft5336'

MaureenHelm commented on samples/drivers/kscan_ft5336/sample.yaml:
```diff
@@ -0,0 +1,12 @@
+        ordered: true
+        regex:
+            - "kb data(.*)"
+    depends_on: kscan
```
You'll want to add `kscan` to the supported list in `boards/arm/stm32f746g_disco/stm32f746g_disco.yaml` to ensure this sample builds for this board in CI. Do it in commit `boards: Configure ft5336 touch panel on stm32f746g_disco board`

k0d replied:
I changed it to kscan_ft5336 so it doesn't run the other kscan sample build.

MaureenHelm reviewed (changes requested):
> @MaureenHelm Could you test the sample on the mimxrt1060_evk board please?

I tested on `mimxrt1050_evk` and it worked after I made a small change:
```diff
diff --git a/boards/arm/mimxrt1050_evk/Kconfig.defconfig b/boards/arm/mimxrt1050_evk/Kconfig.defconfig
index 9763200fef..c338f71ac5 100644
--- a/boards/arm/mimxrt1050_evk/Kconfig.defconfig
+++ b/boards/arm/mimxrt1050_evk/Kconfig.defconfig
@@ -40,7 +40,7 @@ config GPIO_MCUX_IGPIO_5
 endif # GPIO_MCUX_IGPIO

 config I2C
-       default y if LVGL
+       default y if KSCAN

 config I2C_1
        default y
```

k0d committed:
samples: drivers: kscan: Add sample for touch panel drivers

Simple sample to test touch panel drivers.
Outputs a message to the console on every touch event.

Signed-off-by: Mark Olsson <mark@markolsson.se>

k0d committed:
boards: Change dependency for I2C on i.mx rt boards

Changes dependency for I2C so that it's enabled if kscan is enabled.
This is to allow the kscan_ft5336 sample to run.
Note: LVGL has KSCAN as a dependency.
Applies to the mimxrt10{50,60,64}_evk boards.

Signed-off-by: Mark Olsson <mark@markolsson.se>

k0d commented:
> > @MaureenHelm Could you test the sample on the mimxrt1060_evk board please?
> 
> I tested on `mimxrt1050_evk` and it worked after I made a small change:
> 
I've added it in a separate commit and enabled the kscan_ft5336 sample for the 3 imx boards that have the ft5336 chip. I can't test it, but it's such a simple change it will work. I should get one of those boards at some point...

MaureenHelm reviewed (approved):
> I've added it in a separate commit and enabled the kscan_ft5336 sample for the 3 imx boards that have the ft5336 chip. I can't test it, but it's such a simple change it will work. I should get one of those boards at some point...

Thank you! I tested the updated PR again on `mimxrt1050_evk` and it works.

jfischer-phytec-iot commented on samples/drivers/kscan_ft5336/src/main.c:
```diff
@@ -0,0 +1,41 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
```
Please make it generic, the sample should be for the driver API not for specific driver.

k0d replied:
Isn't the other sample specific to another driver? I guess it's better to change it to be a touch sample?

jfischer-phytec-iot replied:
> Isn't the other sample specific to another driver?

Indeed. Could you rename your sample to something like kscan_touch and make it more generic?

erwango commented on boards/arm/mimxrt1050_evk/Kconfig.defconfig:
```diff
@@ -40,7 +40,7 @@ config GPIO_MCUX_IGPIO_5
 
 config I2C
-	default y if LVGL
+	default y if KSCAN
```
What about `if (LVGL || KSCAN)` ?
We might want to enable LVGL w/o KSCAN. This is not what is done below in this file, but this is questionable.
@MaureenHelm, what is your opinion ?

k0d replied:
I'll do whatever you think is best on this one. Lower down in the file is...
```config KSCAN
	default y if LVGL
```
So there is some dependancy chain there.

MaureenHelm replied:
The dependency is for KSCAN; LVGL doesn't use I2C.

k0d replied:
So the question is if we have KSCAN as a dependancy of LVGL?

erwango commented on boards/arm/stm32f746g_disco/stm32f746g_disco.dts:
```diff
@@ -18,6 +18,10 @@
 	};
+	
+	aliases {
+		kscan0 = &touch_controller;
```
Please move this to the existing aliases section below.

k0d replied:
Oops...missed that as it was lower down.

erwango commented on boards/arm/stm32f746g_disco/Kconfig.defconfig:
```diff
@@ -38,4 +45,17 @@ config PWM_STM32_1
 	depends on PWM
 
+config KSCAN
+	default y if LVGL
```
I don't think we should enable KSCAN depending on LVGL (unless there is a dependency in the drivers itself, but ideally it should not be the case.).

k0d replied:
ok, I can agree with that.

erwango reviewed (changes requested):
Looks fine. Few changes requested.
Main concern is the dependency w/ LVGL.

erwango reviewed (approved):
I'm approving the change.

Remaining point is the fact that KSCAN depends on LVGL activation in imx board configuration files, but:
- this is not directly related to this change
- it's up to @MaureenHelm to give the final word on these boards.

k0d commented:
I will change this to use the new DT_INST macros that galek has been working on...then I think this is ready to be merged.

k0d commented:
I've changed this to use the new DT_INST macros that @galak  has been working on...

I'm happy with the changes now and I think this is ready to be merged.

erwango commented:
@k0d ; can you fix shippable ?

k0d committed:
tests: drivers: fixes for failing tests

Tests were failing as they were looking for an alias which didn't exist
Also, because some code in kscan_handlers.c wasn't compilable

Signed-off-by: Mark Olsson <mark@markolsson.se>

MaureenHelm commented:
> @k0d ; can you fix shippable ?

@k0d ping

erwango commented:
@kod, can you rebase ?

k0d commented:
I've rebased...will see if shippable works...otherwise I'll address this later today. I've been working on a different project the past 2 weeks...so haven't had time to look at this until now.

erwango commented on boards/arm/stm32f746g_disco/Kconfig.defconfig:
```diff
@@ -26,6 +26,16 @@ config UART_6
+config I2C
+	default y if KSCAN
+
+config I2C_1
+	default y
+	depends on I2C
+
+config I2C_3
+	default y if KSCAN
+	depends on I2C
```
Please remove these. cf https://github.com/zephyrproject-rtos/zephyr/pull/24342

erwango commented on boards/arm/stm32f746g_disco/pinmux.c:
```diff
@@ -42,6 +42,10 @@ static const struct pin_config pinconf[] = {
 	{STM32_PIN_PB8, STM32F7_PINMUX_FUNC_PB8_I2C1_SCL},
 	{STM32_PIN_PB9, STM32F7_PINMUX_FUNC_PB9_I2C1_SDA},
 #endif  /* CONFIG_I2C_1 */
+#ifdef CONFIG_I2C_3
```
#if DT_HAS_NODE(DT_NODELABEL(i2c3))

k0d commented:
d

k0d commented:
@erwango Shippable fixed, all seems to be good now. Ready to merge I think.

MaureenHelm commented on drivers/kscan/kscan_handlers.c:
```diff
@@ -11,21 +11,21 @@ static inline int z_vrfy_kscan_config(struct device *dev,
 #include <syscalls/kscan_disable_callback_mrsh.c>
 
-static int z_vrfy_kscan_enable_callback(struct device *dev);
+static int z_vrfy_kscan_enable_callback(struct device *dev)
```
The changes in this file don't appear to be related to the commit message, `tests: drivers: kscan fixes failing test due to wrong devicetree define`

k0d replied:
Fixed!

MaureenHelm reviewed (approved):
@erwango are you good with this?

MaureenHelm commented:
Closing and reopening to retrigger CI checks that never returned
